test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Extensions/Binders/ApplyBinderTests.cs

```diff
@@ -165,20 +165,62 @@ public void BindApplyWitGroupByWithNavigationShouldReturnApplyClause()
             groupBy.Kind.Should().Be(TransformationNodeKind.GroupBy);
             groupBy.GroupingProperties.Should().NotBeNull();
             groupBy.GroupingProperties.Should().HaveCount(1);
+            groupBy.ChildTransformations.Should().BeNull();
 
             List<GroupByPropertyNode> groupingProperties = groupBy.GroupingProperties.ToList();
             GroupByPropertyNode dogNode = groupingProperties[0];
-            dogNode.Expression.Should().BeNull();
             dogNode.Name.Should().Be("MyDog");
+            dogNode.Expression.Should().BeNull();
             dogNode.ChildTransformations.Should().HaveCount(1);
 
-            GroupByPropertyNode nameNode = dogNode.ChildTransformations[0];
+            GroupByPropertyNode colorNode = dogNode.ChildTransformations[0];
+            colorNode.Name.Should().Be("Color");
+            colorNode.Expression.Should().BeSameAs(FakeBindMethods.FakePersonDogColorNode);
+            colorNode.ChildTransformations.Should().HaveCount(0);
+        }
+
+        [Fact]
+        public void BindApplyWitGroupByWithDeepNavigationShouldReturnApplyClause()
+        {
+            IEnumerable<QueryToken> tokens = _parser.ParseApply("groupby((MyDog/FastestOwner/MyFavoritePainting/Value))");

```

Comment:
Could you add test to demo that mixing complex types and navigation properties work too? From what I see in ApplyBinder it should work, but to just make sure that it's the case.

Reply:
Added some more tests. Those are the covered scenarios:

_Navigation/SingleValue
Navigation/Navigation/SingleValue
Navigation/Complex/SingleValue
Navigation/Navigation/Complex/SingleValuee_

_Complex/SingleValue
Complex/Complex/SingleValue
Complex/Navigation/SingleValue
Complex/Complex/Navigation/SingleValue_

_Complex/Navigation/Navigation/Complex/SingleValue
Navigation/Complex/Complex/Navigation/SingleValue_

This should be enough?